agent/structs/structs.go

```diff
@@ -1541,6 +1541,9 @@ type CheckServiceNode struct {
 	Node    *Node
 	Service *NodeService
 	Checks  HealthChecks
+
+	// GatewayServices ...
+	GatewayServices GatewayServices `json:"-" bexpr:"-"`
```

Comment:
This addition makes `CheckServiceNode` and `ServiceInfo` equivalent. I wonder if we should just get rid of `ServiceInfo`? Or migrate everything from `CheckServiceNode` to `ServiceInfo`? I am not sure whether struct names are encoded in the RPC responses and whether we need to worry about backwards compatiblity.